ui/selectors/selectors.js

```diff
@@ -1334,13 +1335,59 @@ export function getAllAccountsOnNetworkAreEmpty(state) {
   return hasNoNativeFundsOnAnyAccounts && hasNoTokens;
 }
 
-export function getShouldShowSeedPhraseReminder(state) {
-  const { tokens, seedPhraseBackedUp, dismissSeedBackUpReminder } =
-    state.metamask;
+export function getIsAnyAccountWithBalanceOnNonTestNetwork(state) {
+  const { cachedBalances } = state.metamask;
+
+  const zeroBalanceCheck = (currentValue) =>
+    currentValue === '0x0' || currentValue === '0x00';
+
+  const balancesOnNonTestNetworks = Object.entries(cachedBalances)
+    .filter(
+      (networkAndAccountBalances) =>
+        TEST_CHAINS.includes(networkAndAccountBalances[0]) === false &&
+        networkAndAccountBalances,
+    )
+    .map((accountAndBalance) => Object.values(accountAndBalance[1]))
+    .map((balance) => balance.every(zeroBalanceCheck));
+
+  return balancesOnNonTestNetworks.includes(false);
+}
+
+export async function getIsAnyTokenWithNonZeroBalance(state) {
+  const { tokens } = state.metamask;
+  const selectedAddress = getSelectedAddress(state);
+
+  if (tokens.length > 0) {
+    for (let i = 0; i < tokens.length; i++) {
+      const res = await getERC20Balance(tokens[i], selectedAddress);
```

Comment:
This doesn't have to be `await`ed, these checks could run in parallel.  I'm not sure how important that is because I'm not sure how long each one takes.

Reply:
Fixed

Reply:
Sorry, I guess I said this in a confusing way.  The change you made doesn't work, because `res` is now a Promise.

You need to do something a lot fancier, using something like `Promise.all()` https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all

Reply:
Yes, I misunderstood you...
I fixed it here: 0f0ceebd9f6b1c8b560ecd1dc2172edc7cbe2bbe
Also, I had to revert some changes - `shouldShowSeedPhraseReminder: PropTypes.oneOfType([PropTypes.bool, PropTypes.object]).isRequired`. That's because `shouldShowSeedPhraseReminder` is a `Promise`  at one moment before it gets resolved...